purchase_sale_stock_inter_company/models/stock_picking.py

```diff
@@ -19,26 +19,61 @@ def action_done(self):
             purchase = pick.sale_id.auto_purchase_order_id
             if not purchase:
                 continue
-            purchase.picking_ids.write({"intercompany_picking_id": pick.id})
-            if not pick.intercompany_picking_id and purchase.picking_ids[0]:
-                pick.write({"intercompany_picking_id": purchase.picking_ids[0]})
+            po_picking_pending = purchase.picking_ids.filtered(
+                lambda x: x.state not in ["done", "cancel"]
+            )
+            po_picking_pending.write({"intercompany_picking_id": pick.id})
+            if not pick.intercompany_picking_id and po_picking_pending[0]:
+                pick.write({"intercompany_picking_id": po_picking_pending[0]})
             for move in pick.move_lines:
                 move_lines = move.move_line_ids
-                po_move_lines = move.sale_line_id.auto_purchase_line_id.move_ids.filtered(
+                po_moves_pending = move.sale_line_id.auto_purchase_line_id.move_ids.filtered(
                     lambda x, ic_pick=pick.intercompany_picking_id: x.picking_id
                     == ic_pick
-                ).mapped(
-                    "move_line_ids"
+                    and x.state not in ["done", "cancel"]
                 )
-                if not len(move_lines) == len(po_move_lines):
-                    raise UserError(
-                        _(
-                            "Mismatch between move lines with the "
-                            "corresponding  PO %s for assigning "
-                            "quantities and lots from %s for product %s"
+                po_move_lines = po_moves_pending.mapped("move_line_ids")
+                if move.product_id.tracking == "serial":
```

Comment:
Why doing a special case depending on the product tracking instead of just mimicking the `stock.move.line`?

Reply:
> Why doing a special case depending on the product tracking instead of just mimicking the `stock.move.line`?

My error occurs only with serial products, so I isolated the error. In other cases, the check is fine, but for serial products, it’s not.

Reply:
I don't think so, as lots may have the same problem if you have more than one lot. As said, you should mimic the `stock.move.line` in one place and the other.

Reply:
Done, please update your review